lxd/cluster/recover.go

```diff
@@ -299,6 +306,111 @@ func writeRecoveryTarball(databaseDir string, raftNodes []db.RaftNode) (string,
 	return tarballPath, nil
 }
 
+// UnpackRecoveryTarball unpacks the tarball found at `tarballPath`, replaces
+// the global database, updates the local database with any changed addresses,
+// and writes a global patch file to update the global database with any changed
+// addresses.
+func UnpackRecoveryTarball(tarballPath string, database *db.Node) error {
```

Comment:
I think the name of this function could be clearer. "Unpack" sounds harmless enough and not destructive, but reading the description it says "replaces the global database" which is very impactful.

So how about "DatabaseReplaceFromTarball"?

Reply:
Out of curiosity, why `DatabaseReplace` rather than `ReplaceDatabase`? I've typically started with a verb.